### PR TITLE
Internal features to facilitate debugging

### PR DESCRIPTION
**Pull request by marcosholgado**
Task/Issue URL: https://app.asana.com/0/488551667048375/1206159199938929/f 

### Description
See task

### Steps to test this PR
See task

### CODE REVIEW

**Comment by marcosholgado** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #4002** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4002?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/4002?utm_source=stack-comment).

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt [hunk 1]
```diff
@@ -168,6 +173,15 @@ class RealSubscriptionsManager @Inject constructor(
         }
     }
 
+    override suspend fun deleteAccount(): Boolean {
```
Comment (marcosholgado) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt:176]: This is just for internal feature so not testing this method yet.

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt [hunk 2]
```diff
@@ -359,7 +373,11 @@ class RealSubscriptionsManager @Inject constructor(
                             logcat(LogPriority.DEBUG) { "Subs: auth token expired" }
                             val subscriptionsData = recoverSubscriptionFromStore()
                             if (subscriptionsData is Success) {
-                                AuthToken.Success(authDataStore.authToken!!)
+                                return if (subscriptionsData.entitlements.isEmpty()) {
```
Comment (marcosholgado) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt:376]: This was a bug where we weren't checking entitlements after refreshing an expired token so it looked like you had a subscription for a second when you didn't.

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt [hunk 3]
```diff
@@ -119,6 +119,11 @@ interface SubscriptionsManager {
      * Signs the user out
      */
     suspend fun signOut()
+
+    /**
+     * Deletes the current account
+     */
```
Comment (aitorvs) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt:125]: nit: Add docs for the returned value

**Review by aitorvs — APPROVED** [review]

**Pull request merged**